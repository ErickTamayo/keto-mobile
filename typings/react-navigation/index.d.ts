/* eslint-disable */
import ReactNavigation from 'react-navigation'

declare module 'react-navigation' {
  export interface StackNavigatorConfig {
    cardShadowEnabled: boolean
  }
}
