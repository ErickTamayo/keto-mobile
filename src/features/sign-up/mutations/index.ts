import gql from 'graphql-tag'

export const SET_SIGN_UP_USER_GENDER = gql`
  mutation SetGender($gender: String!) {
    setGender(gender: $gender) @client
  }
`

export const SET_SIGN_UP_USER_OTHER_PARAMETERS = gql`
  mutation setOtherParametersToSignUpUser(
    $age: String!
    $height: Height!
    $weight: Weight!
    $weightGoal: Weight!
  ) {
    setOtherParametersToSignUpUser(
      age: $age
      height: $height
      weight: $weight
      weightGoal: $weightGoal
    ) @client
  }
`
