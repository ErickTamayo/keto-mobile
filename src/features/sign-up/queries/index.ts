import gql from 'graphql-tag'

export const GET_NEW_USER = gql`
  {
    newUser @client {
      displayName
      email
      password
      gender
      age
      height {
        unit
        feet
        inches
        centimeters
      }
      weight {
        unit
        pounds
        kilograms
      }
      weightGoal {
        unit
        pounds
        kilograms
      }
    }
  }
`
