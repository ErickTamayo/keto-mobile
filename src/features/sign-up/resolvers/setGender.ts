import gql from 'graphql-tag'
import { SIGN_UP_USER_TYPENAME } from '../constants'

const setGender = (_, { gender }, { cache, getCacheKey }) => {
  const id = getCacheKey({ __typename: SIGN_UP_USER_TYPENAME, id: '0' })

  const fragment = gql`
    fragment selectGender on SignUpUser {
      gender
    }
  `

  cache.writeFragment({ id, fragment, data: { gender, __typename: SIGN_UP_USER_TYPENAME } })
  return null
}

export default setGender
