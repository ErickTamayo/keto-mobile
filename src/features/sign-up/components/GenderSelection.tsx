import * as React from 'react'
import { Text, View } from 'react-native'
import { NavigationInjectedProps } from 'react-navigation'
import { StateProps, DispatchProps } from '../containers/GenderSelection'
import st from '../../../styles'
import CardWithHeaderAndButton from '../../../components/card-with-header-and-button/CardWithHeaderAndButton'

export interface Props extends StateProps, DispatchProps, NavigationInjectedProps {}

export default class GenderSelection extends React.Component<Props, object> {
  private navigateToOtherParameters = () => {
    const { navigation } = this.props
    navigation.navigate('OtherParameters')
  }

  // private navigateToLogIn = () => this.props.navigation.navigate('Login')

  public render(): JSX.Element {
    // const { counter, addCounter } = this.props
    const disabled = false
    return (
      <View style={[st.flex.f1, st.items.center, st.justify.center, st.bg.greyLightest]}>
        <CardWithHeaderAndButton
          title="What's your gender?"
          description="We will make sure you get better and personalized results"
          buttonText="Continue"
          onButtonPress={this.navigateToOtherParameters}
          disabled={disabled}
        >
          <Text>Radio Buttons here</Text>
          <Text>Radio Buttons here</Text>
          <Text>Radio Buttons here</Text>
          <Text>Radio Buttons here</Text>
          <Text>Radio Buttons here</Text>
          <Text>Radio Buttons here</Text>
          {/* <RadioButton variant="lg" checked={true} value="female" onChange={this.handleChange}>
            Female
          </RadioButton>
          <RadioButton variant="lg" checked={false} value="male" onChange={this.handleChange}>
            Male
          </RadioButton> */}
        </CardWithHeaderAndButton>
      </View>
    )
  }
}
