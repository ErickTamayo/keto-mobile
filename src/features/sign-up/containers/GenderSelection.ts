import { connect } from 'react-redux'
import { Dispatch } from 'redux'
import { State } from '../records/SignUpState'
import GenderSelection from '../components/GenderSelection'
import { selectGender, SelectGender } from '../actions'

export interface StateProps {
  gender: 'male' | 'female' | null
}

export interface DispatchProps {
  selectGender: (gender: 'male' | 'female') => SelectGender
}

export const mapStateToProps = ({ signUp }: { signUp: State }): StateProps => {
  return {
    gender: signUp.gender,
  }
}

export const mapDispatchToProps = (dispatch: Dispatch): DispatchProps => ({
  selectGender: (gender: 'male' | 'female'): SelectGender => dispatch(selectGender(gender)),
})

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(GenderSelection)
