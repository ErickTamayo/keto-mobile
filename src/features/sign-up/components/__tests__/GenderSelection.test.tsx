import * as React from 'react'
import { TouchableOpacity, GestureResponderEvent } from 'react-native'
import { mount } from 'enzyme'
import { NavigationScreenProp, NavigationRoute, NavigationParams } from 'react-navigation'
import { MockedProvider } from 'react-apollo/test-utils'
import wait from 'waait'
import { Mutation } from 'react-apollo'

import { GET_SIGN_UP_USER_GENDER } from '../../queries'
import { SET_SIGN_UP_USER_GENDER } from '../../mutations'

import GenderSelection, { Props } from '../GenderSelection'
import setGender from '../../resolvers/setGender'

describe('<GenderSelection />', () => {
  it('renders correctly', async () => {
    const props: Props = {
      navigation: {} as NavigationScreenProp<NavigationRoute<NavigationParams>, NavigationParams>,
      gender: null,
      setGender: jest.fn(),
    }

    const mocks = [
      {
        request: {
          query: GET_SIGN_UP_USER_GENDER,
          variables: {},
        },
        result: {
          data: {
            signUpUser: { gender: null },
          },
        },
      },
    ]

    const wrapper = mount(
      <MockedProvider mocks={mocks} addTypename={false}>
        <GenderSelection {...props} />
      </MockedProvider>
    )

    await wait(0)
    wrapper.update()

    expect(wrapper.find(GenderSelection)).toMatchSnapshot()
  })

  it('selects female on press the first radio button', async () => {
    const props: Props = {
      navigation: {} as NavigationScreenProp<NavigationRoute<NavigationParams>, NavigationParams>,
      gender: null,
      setGender: jest.fn(),
    }

    const mocks = [
      {
        request: {
          query: SET_SIGN_UP_USER_GENDER,
          variables: { gender: 'female' },
        },
        result: {
          data: {
            setGender: { id: '0', gender: 'female' },
          },
        },
      },
      {
        request: {
          query: GET_SIGN_UP_USER_GENDER,
          variables: {},
        },
        result: {
          data: {
            signUpUser: { gender: null },
          },
        },
      },
    ]

    const wrapper = mount(
      <MockedProvider mocks={mocks} addTypename={false}>
        <GenderSelection {...props} />
      </MockedProvider>
    )

    await wait(0)
    wrapper.update()

    expect(props.setGender).not.toBeCalled()

    wrapper
      .find(TouchableOpacity)
      .at(0)
      .prop('onPress')!({} as GestureResponderEvent)

    await wait(0)
    wrapper.update()

    expect(props.setGender).toBeCalledWith({ variables: { gender: 'female' } })
    expect(props.setGender).toBeCalled()
  })

  it('selects male on press the second radio button', async () => {
    const props: Props = {
      navigation: {} as NavigationScreenProp<NavigationRoute<NavigationParams>, NavigationParams>,
      gender: null,
      setGender: jest.fn(),
    }

    const mocks = [
      {
        request: {
          query: SET_SIGN_UP_USER_GENDER,
          variables: { gender: 'female' },
        },
        result: {
          data: {
            setGender: { id: '0', gender: 'female' },
          },
        },
      },
      {
        request: {
          query: GET_SIGN_UP_USER_GENDER,
          variables: {},
        },
        result: {
          data: {
            signUpUser: { gender: null },
          },
        },
      },
    ]

    const wrapper = mount(
      <MockedProvider mocks={mocks} addTypename={false}>
        <GenderSelection {...props} />
      </MockedProvider>
    )

    await wait(0)
    wrapper.update()

    expect(props.setGender).not.toBeCalled()

    wrapper
      .find(TouchableOpacity)
      .at(1)
      .prop('onPress')!({} as GestureResponderEvent)

    await wait(0)
    wrapper.update()

    expect(props.setGender).toBeCalledWith({ variables: { gender: 'male' } })
    expect(props.setGender).toBeCalled()
  })
})
