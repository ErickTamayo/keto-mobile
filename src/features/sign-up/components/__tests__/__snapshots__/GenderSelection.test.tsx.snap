// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GenderSelection /> renders correctly 1`] = `
<Component
  style={
    Array [
      Object {
        "flex": 1,
      },
      Object {
        "alignItems": "center",
      },
      Object {
        "justifyContent": "center",
      },
      Object {
        "backgroundColor": "#f7f7f7",
      },
    ]
  }
>
  <CardWithHeaderAndButton
    buttonText="Continue"
    description="We will make sure you get better and personalized results"
    disabled={true}
    onButtonPress={[Function]}
    title="What's your gender?"
  >
    <Component
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
        ]
      }
    >
      <RadioButton
        checked={false}
        onPress={[Function]}
        style={
          Array [
            Object {
              "margin": 26,
            },
          ]
        }
        title="FEMALE"
        value="female"
        variant="lg"
      />
      <RadioButton
        checked={false}
        onPress={[Function]}
        style={
          Array [
            Object {
              "margin": 26,
            },
          ]
        }
        title="MALE"
        value="male"
        variant="lg"
      />
    </Component>
  </CardWithHeaderAndButton>
</Component>
`;
